src/server/pfs/server/driver_new_storage.go

```diff
@@ -0,0 +1,219 @@
+package server
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"path"
+	"strconv"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/auth"
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	pfsserver "github.com/pachyderm/pachyderm/src/server/pfs"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/chunk"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset/index"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset/tar"
+	txnenv "github.com/pachyderm/pachyderm/src/server/pkg/transactionenv"
+	"github.com/pachyderm/pachyderm/src/server/pkg/work"
+	"golang.org/x/net/context"
+)
+
+const (
+	compacted = "compacted"
+)
+
+func init() {
+	// This is kind of weird, but I think it might be necessary.
+	// Since we build our protos with gogoproto, we need to register
+	// the types that can be embedded in an Any type with the gogoproto
+	// proto package because building the protos just registers them with the
+	// normal proto package.
```

Comment:
This does seem a bit weird, I'd expect to be able to do this in the proto files. Not really a big deal though, I don't really see a way this could go wrong.

Reply:
Yeah, this threw me off for a bit. I have to use the gogoproto types package to use Any because the regular ptypes Any does not have the methods used by gogoproto to speed up serialization/deserialization. But, our proto types do not get registered with the gogoproto proto package during our compilation process, so it does not know how to serialize/deserialize the type unless I manually register them. I'm not sure I fully understand why this is the case, and if it means there should be some changes to our proto compilation or if I am just missing something.

Reply:
Makes sense, I feel like if this works then might as well just do it until we figure out a better way to do it. Even if it's weird, it's not weird in a way that seems like to cause confusion later on.